configure/CMakeLists.txt

```diff
@@ -198,6 +198,7 @@ if(NOT WIN32)
     # throw specifications and these are not supported anymore.
     CHECK_CXX_COMPILER_FLAG("-std=c++14" COMPILER_SUPPORTS_CXX14)
     CHECK_CXX_COMPILER_FLAG("-Og" COMPILER_SUPPORTS_OG)
+    CHECK_CXX_COMPILER_FLAG("-g" COMPILER_SUPPORTS_G)
```

Comment:
Can we remove these checks and assume the flags are always available?

GCC: supports '-Og -g' since 4.8 (Debian 8 - oldest OS we support in CI has GCC 4.9)
Clang: supports '-Og -g' since 4.0.0

Reply:
I don't think assuming that it is present is good practice. What happens if someone tries to compile it with other compilers (Intel/HP UX, etc)?

Reply:
I'm correcting myself, this is in the GNU/CLANG branch. Will do.